File: hacker_rank/PALG-1-solve-me-first/python/main.py
#!/bin/python3

import sys


def parse_input(text):
    lines = text.split('\n')
    x = int(lines[0])
    y = int(lines[1])
    return x, y


def main(x, y):
    return x + y


if __name__ == '__main__':
    args = parse_input(sys.stdin.read())
    print(main(*args))

File: hacker_rank/PALG-1-solve-me-first/rust/src/main.rs
use std::io;
use std::io::prelude::*;
use std::num::ParseIntError;


fn parse_input() -> Result<(i32, i32), ParseIntError> {
    let stdin = io::stdin();
    let mut lines: Vec<String> = stdin
        .lock()
        .lines()
        .take(2)
        .filter_map(|line| line.ok())
        .collect();

    let x: i32 = lines.remove(0).parse()?;
    let y: i32 = lines.remove(0).parse()?;

    Ok((x, y))
}


fn main() {
    match parse_input() {
        Ok((x, y)) => println!("{}", x + y),
        Err(_) => println!("Unable to parse inputs"),
    }
}
